Frontend/nutri-pal/src/app/diary/page.tsx

```diff
@@ -0,0 +1,223 @@
+    const monthsOptions = [
+        { value: '', label: 'Filter by Month' },
+        { value: '01', label: 'January' },
+        { value: '02', label: 'February' },
+        { value: '03', label: 'March' },
+        { value: '04', label: 'April' },
+        { value: '05', label: 'May' },
+        { value: '06', label: 'June' },
+        { value: '07', label: 'July' },
+        { value: '08', label: 'August' },
+        { value: '09', label: 'September' },
+        { value: '10', label: 'October' },
+        { value: '11', label: 'November' },
+        { value: '12', label: 'December' }
+    ]
+
+    // Function tu turn the ISO date into "January 01, 2024" style
+    const isoToDate = (iso: string) => {
+        const dateArray = iso.split('-')
+
+        return (`${months_dictionary[parseInt(dateArray[1])]} ${dateArray[2].slice(0, 2)}, ${dateArray[0]}`)
+    }
+
+    // Setting the user that was authenticated by Firebase to a variable and declare the flag to start making request to true
+    onAuthStateChanged(auth, (currentUser) => {
+        if (currentUser) {
+            setUser(currentUser);
+            setGetFlag(true)
+        }
+    })
+
+    // Do a GET request for the registered days that the user has with or without the query params
+    const getDates = (year = '', month = '', day = '') => {
+        const HISTORY_DAYS = BASE_URL_REST_API + 'history/registered_days/' + user.uid + `?year=${year}&month=${month}&day=${day}`;
+        axios.get(HISTORY_DAYS)
+            .then((response) => { setDates(response.data) })
+            .catch((error) => { alert('There has been an error trying to get the dates of the user \n' + error) })
+    }
+
+    // Do a GET request to get the maximum and minimum registered days the user has
+    const getLimitDates = () => {
+        const MAX_MIN_DATES = BASE_URL_REST_API + 'history/maxmindate/' + user.uid;
+        axios.get(MAX_MIN_DATES)
+            .then((response) => { setMaxDate(response.data._max.date.slice(0, 4)); setMinDate(response.data._min.date.slice(0, 4)) })
+            .catch((error) => { alert('There has been an error trying to get the maximum and minimum dates of the user \n' + error) })
+    }
+
+    // After auth make the GET requests to display the cards
+    useEffect(() => {
+        if (getFlag) {
+            getDates()
+            getLimitDates()
+        }
+    }, [getFlag])
+
+    // Function to open and closing the modal
+    const toggleModal = () => {
+        if (openModal) {
+            setOpenModal(false)
+        } else {
+            setOpenModal(true)
+        }
+    }
+
+    // When the input calendar date is changed then reset the values of the selected year and month filter, and then separate the string into year month and date for the GET request
+    useEffect(() => {
+        setFilterYear('')
+        setFilterMonth('')
+        if (filterDate.length > 0) {
+            const temp = filterDate.split('-');
```

Comment:
let's call this dateString or something like that; for style points we could do something like

const [year, month, day] = filterDate.split('-'); 

https://budiirawan.com/typescript-destructuring-array/